### PR TITLE
Unified service collections

### PR DESCRIPTION
**Pull request by mmadariaga**
- Use a single service collection instance for every binded event (We were using a instance per event: on commit, pre remove, etc).

- Unify how common and entity specific lifecycle services are registered.

- List binded services on service collections (Visual sugar, it's overridden on cache compilation and it will not show customer specific services).

### CODE REVIEW

File: library/Ivoz/Core/Infrastructure/Symfony/DependencyInjection/Compiler/LifecycleCompiler.php [hunk 1]
```diff
@@ -156,45 +126,7 @@ protected function getLifecycleServices(): array
...
-        return $services;
-    }
-
-    /**
-     * @return Definition[]
-     *
-     * @psalm-return array<int, Definition>
-     */
-    protected function getDomainEventSubscriberServices(): array
+    protected function getDomainEventSubscriberServices()
```
Comment (Kaian) [inline library/Ivoz/Core/Infrastructure/Symfony/DependencyInjection/Compiler/LifecycleCompiler.php]: Keep return type for getDomainEventSubscriberServices

File: library/Ivoz/Core/Domain/Service/LifecycleEventHandlerInterface.php [hunk 2]
```diff
@@ -19,7 +20,8 @@ interface LifecycleEventHandlerInterface
         self::EVENT_POST_PERSIST,
         self::EVENT_PRE_REMOVE,
         self::EVENT_POST_REMOVE,
-        self::EVENT_ON_COMMIT
+        self::EVENT_ON_COMMIT,
+        self::EVENT_ON_ERROR
```
Comment (Kaian) [inline library/Ivoz/Core/Domain/Service/LifecycleEventHandlerInterface.php]: Add comma at the end of the array entry

File: library/Ivoz/Core/Domain/Service/LifecycleServiceCollectionInterface.php [hunk 3]
```diff
@@ -6,11 +6,18 @@
...
 {
-    public function setServices(array $services);
+    public function setServices(string $event, array $services);
 
     /**
      * @param EntityInterface $entity
      * @return mixed
      */
-    public function execute(EntityInterface $entity);
+    public function execute(string $event, EntityInterface $entity);
```
Comment (Kaian) [inline library/Ivoz/Core/Domain/Service/LifecycleServiceCollectionInterface.php]: Function comment misses $event parameter

File: library/Ivoz/Core/Domain/Service/LifecycleServiceCollectionTrait.php [hunk 4]
```diff
@@ -21,30 +21,77 @@ public function __construct(
...
+    {
+        $event = LifecycleEventHandlerInterface::EVENT_ON_ERROR;
+        $this->services[$event][] = $service;
+    }
 
     /**
      * @param EntityInterface $entity
      */
-    public function execute(EntityInterface $entity)
+    public function execute(string $event, EntityInterface $entity)
```
Comment (Kaian) [inline library/Ivoz/Core/Domain/Service/LifecycleServiceCollectionTrait.php]: Function comment misses $event parameter

**Comment by ironArt3mis** [comment]
Rebase completed.

**Comment by ironArt3mis** [comment]
Rebase completed.

**Pull request merged**